### DISCUSSION TITLE
Catastrophic failure in response to Xaml error.

### DISCUSSION THREAD

**Discussion by poepoe12002** [unoplatform/uno#11619] [Q&A]
The following is from a WinUI 3 project.  
Visual Studio behaves correctly regardless of the correctness of the xaml. 
Depending on which xaml is un-commented, either the project will build and the program will run or the build will fail with errors and warnings.

<Window
    x:Class="WinUI3.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:local="using:WinUI3"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">

    <NavigationView
        IsBackButtonVisible="Collapsed"
        IsPaneToggleButtonVisible="False"
        IsSettingsVisible="False"
        PaneDisplayMode="Left">

        
        
        <NavigationView.PaneTitle>PaneTitle</NavigationView.PaneTitle>
        
        

        
        
        
        
        
    </NavigationView>
</Window>

The same code in an Uno Project behaves quite differently.  Where the correct xaml is un-commented, the project will build and the program will run.  However, under the scenario with the erroneous xaml, the build grinds to a halt.  It will not finish, so no errors and warnings.  Visual Studio then has to be killed and the OS becomes unstable.

<Page
    x:Class="UPA.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:UPA"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <NavigationView
            IsBackButtonVisible="Collapsed"
            IsPaneToggleButtonVisible="False"
            IsSettingsVisible="False"
            PaneDisplayMode="Left">

        
        

        

        
        

        
        

        

        
        
    </NavigationView>
</Page>

Thanks

**Comment by jeromelaban** [comment]
Thanks. Can you provide a standalone repro app so we can make sure we're looking at the same issue?

**Reply by poepoe12002** [reply]
I can upload a repro.  It will be a solution with a normal WinUI 3 project and a Uno Platform App with multiple heads.  Would that be ok.  If so, should I clean the solution first?

**Reply by poepoe12002** [reply]
Here it is: https://github.com/poepoe12002/NavigationViewTest

**Reply by jeromelaban** [reply]
I just noticed, this is not valid XAML:
```xml
    <NavigationView.PaneTitle
        FontSize="16"
        FontWeight="Bold"
        Text="PaneTitle" />
```
This may be a very good case of visual studio falling into some unsupported pattern or loop. [`PaneTitle` is a string](https://learn.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.navigationview.panetitle?view=winui-2.8), not a complex property that you can style this way.

Can you try the following:
```xml
<NavigationView.PaneTitle>My Pane Title</NavigationView.PaneTitle>
```
And see if it helps?

Also if you want to change the appearance the `PaneTitle`, you'll need to override the style of [the default style of NavigationView](https://github.com/unoplatform/uno/blob/d05305075f54545174d7e4ca95f6edfcdc39d807/src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/NavigationView.xaml#L12) in your app.

If it helps, we'll likely need to open an issue on Microsoft's dev community with your sample for their team to fix.

**Reply by poepoe12002** [reply]
Hopefully you able to reproduce the VS failure regarding this xaml in a Uno Platform project.  I only bring this to your attention because in a Regular WinUI3 project, on my computer, the xaml is dealt with correctly.  No VS failure.  
Well, I'm moving on.  Thanks.

**Reply by jeromelaban** [reply]
Thanks. I'm not able to reproduce, but does having proper XAML avoid the issue?

**Reply by poepoe12002** [reply]
Yes.  Thank you.
